apps/agreement/contracts/Agreement.sol

```diff
@@ -315,21 +315,23 @@ contract Agreement is IAgreement, AragonApp {
     /**
     * @notice Close action #`_actionId`
     * @dev If allowed by the originating Disputable app, this function allows users to close actions that are not:
-    *      - Closed
     *      - Currently challenged
     *      - Ruled as voided
     *      - Ruled in favour of the submitter
+    *      It ignores if the action is already closed, without re-closing it
     *      Initialization check is implicitly provided by `_canClose()` as disputable actions can be created only
     *      via `newAction()` which already requires initialization implicitly through `activate()`
     * @param _actionId Identification number of the action to be closed
     */
     function closeAction(uint256 _actionId) external {
         Action storage action = _getAction(_actionId);
-        require(_canClose(action), ERROR_CANNOT_CLOSE_ACTION);
+        require(_canClose(action, true), ERROR_CANNOT_CLOSE_ACTION);
```

Comment:
I find the `ignoreIfClosed` parameter a bit difficult to understand, partly because its name is a somewhat vague.

However, I also wonder if we shouldn't just move this require down into the `if (!action.closed)` block, and avoid all the complexity. If the action is closed, we don't need to do any more validation on it other than to skip it.

Reply:
Addressed this in the last commit, let me know your thoughts. Note that I had to change some tests since it affects some related behavior